src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -2863,8 +2946,22 @@ def update_mc_profile_preview(self) -> ManagedCluster:
         mc = self.update_storage_profile(mc)
         # update workload auto scaler profile
         mc = self.update_workload_auto_scaler_profile(mc)
+        # update azure monitor metrics profile
+        mc = self.update_azure_monitor_profile(mc)
         # update vpa
         mc = self.update_vpa(mc)
         # update creation data
         mc = self.update_creation_data(mc)
+
+        if (self.__raw_parameters.get("enable_azuremonitormetrics") or self.__raw_parameters.get("disable_azuremonitormetrics")):
```

Comment:
Better to place this in `update_azure_monitor_profile`, it seems this `ensure` function is highly correlated with updating the profile.

Reply:
Updated